build.psm1

```diff
@@ -1392,71 +1392,70 @@ function Start-PSPester {
         Switch-PSNugetConfig -Source Public
     }
 
-    if (-not (Get-Module -ListAvailable -Name $Pester -ErrorAction SilentlyContinue | Where-Object { $_.Version -ge "4.2" } ))
-    {
-        Restore-PSPester
+    #The PowerShell Pester tests have not yet been fully updated to Pester 5.
+    #Update when https://github.com/PowerShell/PowerShell/issues/12816 is resolved.
+    [Microsoft.PowerShell.Commands.ModuleSpecification]$requiredPester = @{
+        ModuleName     = 'Pester';
+        ModuleVersion  = '4.2';
+        MaximumVersion = '4.99'
     }
 
-    if ($IncludeFailingTest.IsPresent)
-    {
+    if (-not (Get-Module -ListAvailable -FullyQualifiedName $requiredPester)) {
+        $publishModulePath = Join-Path $BinDir 'Modules'
+        #Create the base directory if it doesn't exist
+        if (-not (Test-Path $publishModulePath)) {
+            $null = New-Item -Path $publishModulePath -ItemType Directory -Force
```

Comment:
If this directory does not exist, then the project was not built at all. In this case, it is best to end the execution with a useful message.